composer/trainer/trainer.py

```diff
@@ -1343,6 +1348,9 @@ def fit(
             _validate_precision(precision, self._device, self.deepspeed_enabled)
             self.state.precision = precision
 
+            # update scaler since precision was provided
+            self.state.scaler = ClosureGradScaler() if self._use_closures() else GradScaler()
```

Comment:
This would wipe the grad scaler state if one sets the precision in `fit()`, even if it is the same as in `__init__`. Should this line be guarded with a `if precision != state.precision:` check?